### PR TITLE
[somfytahoma] Added support for the Hitachi Yutaki devices

### PR DESCRIPTION
**Pull request by octa22**
This PR adds support for the Hitachi Yutaki devices
- HitachiAirToWaterMainComponent (https://github.com/openhab/openhab-addons/issues/11683)
- HitachiAirToWaterHeatingZone (https://github.com/openhab/openhab-addons/issues/11684)
- HitachiDHW (https://github.com/openhab/openhab-addons/issues/11685)

All three things has been tested by the issue creator @wolfgangroller

Signed-off-by: Ondrej Pecta <opecta@gmail.com>

### CODE REVIEW

**Comment by lolodomo** [comment]
I will review that one.

File: bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/thing/channels.xml [hunk 1]
```diff
@@ -506,5 +507,347 @@
...
+				<option value="heat demand-off">Heat demand off</option>
+				<option value="heat thermo-off">Heat thermo off</option>
+				<option value="heat thermo-on">Heat thermo on</option>
+				<option value="off">Off</option>
+				<option value="swp-off">Swp off</option>
+			</options>
+		</state>
+	</channel-type>
+
+	<channel-type id="outdoor_temp">
```
Comment (lolodomo) [inline bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/thing/channels.xml:757]: Maybe you can rather use the system channel.

File: bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/thing/channels.xml [hunk 2]
```diff
@@ -506,5 +507,347 @@
 		<description>Away Mode (ON or OFF)</description>
 	</channel-type>
 
+	<channel-type id="zone_mode">
```
Comment (lolodomo) [inline bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/thing/channels.xml:510]: Please update the default translations (use the maven command for an automatic update).

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaHitachiATWHZHandler.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        stateNames.put(YUTAKI_TARGET_MODE, "modbus:YutakiTargetModeState");
+        stateNames.put(YUTAKI_MODE, "modbus:YutakiModeState");
+        stateNames.put(HOLIDAY_MODE, "modbus:HolidayModeZone1State");
+        stateNames.put(ALARM_NUMBER, "modbus:AlarmNumberState");
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        super.handleCommand(channelUID, command);
+        if (command instanceof StringType) {
```
Comment (lolodomo) [inline bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaHitachiATWHZHandler.java:46]: Please catch the REFRESH command in your 3 new handlers like you did in other handlers.

File: bundles/org.openhab.binding.somfytahoma/README.md [hunk 4]
```diff
@@ -60,82 +61,116 @@ Please see the example below.
...
+| hitachi (yutaki) air to water main component                                  | liquid_temp_thmi                | current liquid temperature THMI                                                                                                                                                                                                   |
+| hitachi (yutaki) air to water main component                                  | operation                       | current operation state (Alarm, Off, Heat/Cool thermo on, Heat/Cool thermo off, DHW off, Heat/Cool demand off, Swp off)                                                                                                           |
+| hitachi (yutaki) air to water main component                                  | outdoor_temp                    | current oudoor temperature in                                                                                                                                                                                                     |
```
Comment (lolodomo) [inline bundles/org.openhab.binding.somfytahoma/README.md:164]: Outdoor
The final "in" To be removed

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/thing/channels.xml (hunk 1 above)
Comment (octa22) [reply bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/thing/channels.xml:757]: Please can you be more specific? I am sorry, but I have no idea what you mean...

File: bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/thing/channels.xml (hunk 2 above)
Comment (octa22) [reply bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/thing/channels.xml:510]: I have never used it before and I am getting error ([ERROR] No plugin found for prefix 'i18n' in the current project) when I run 
```mvn i18n:generate-default-translations```
Java 11, maven 3.6.0
Any idea what I am doing wrong? \
Thanks

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaHitachiATWHZHandler.java (hunk 3 above)
Comment (octa22) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaHitachiATWHZHandler.java:46]: OK

File: bundles/org.openhab.binding.somfytahoma/README.md (hunk 4 above)
Comment (octa22) [reply bundles/org.openhab.binding.somfytahoma/README.md:164]: Fixed

File: bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/thing/channels.xml (hunk 2 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/thing/channels.xml:510]: Please try : mvn org.openhab.core.tools:i18n-maven-plugin:3.2.0:generate-default-translations

Comment (octa22) [reply bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/thing/channels.xml:510]: solved with the 3.2.0 version

File: bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/i18n/somfytahoma.properties [hunk 5]
```diff
@@ -133,9 +166,25 @@ channel-type.somfytahoma.derogation_heating_mode.description = Derogation heatin
 channel-type.somfytahoma.derogation_heating_mode.command.option.auto = Program
 channel-type.somfytahoma.derogation_heating_mode.command.option.away = Away
 channel-type.somfytahoma.derogation_heating_mode.command.option.comfort = Home
-channel-type.somfytahoma.derogation_heating_mode.command.option.frost\ protection = Frost protection
+channel-type.somfytahoma.derogation_heating_mode.command.option.frost protection = Frost protection
```
Comment (lolodomo) [inline bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/i18n/somfytahoma.properties:169]: Please restore the backslash, this is a bug in the tool.

File: bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/i18n/somfytahoma.properties [hunk 6]
```diff
@@ -133,9 +166,25 @@ channel-type.somfytahoma.derogation_heating_mode.description = Derogation heatin
...
 channel-type.somfytahoma.derogation_heating_mode.command.option.manual = Manual
 channel-type.somfytahoma.derogation_heating_mode.command.option.eco = Night
+channel-type.somfytahoma.dhw.label = DHW Control
+channel-type.somfytahoma.dhw.description = Domestic hot water control
+channel-type.somfytahoma.dhw.state.option.run = Run
+channel-type.somfytahoma.dhw.state.option.stop = Stop
+channel-type.somfytahoma.dhw_mode.label = DHW Mode
+channel-type.somfytahoma.dhw_mode.description = DHW mode control
+channel-type.somfytahoma.dhw_mode.state.option.standard = Standard
+channel-type.somfytahoma.dhw_mode.state.option.high demand = High Demand
```
Comment (lolodomo) [inline bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/i18n/somfytahoma.properties:179]: Please add a backslash before the space on the key. 
Please fix all other entries.

File: bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/i18n/somfytahoma.properties [hunk 7]
```diff
@@ -197,8 +252,22 @@ channel-type.somfytahoma.open_closed_valve.label = Valve Open/Closed state
 channel-type.somfytahoma.open_closed_valve.description = Current open/closed state of the Somfy thermostatic valve
 channel-type.somfytahoma.operating_mode.label = Operating mode
 channel-type.somfytahoma.operating_mode.description = Operating mode of the Somfy thermostatic valve
+channel-type.somfytahoma.operation.label = Operation State
+channel-type.somfytahoma.operation.description = Operation state
+channel-type.somfytahoma.operation.state.option.alarm = Alarm
+channel-type.somfytahoma.operation.state.option.cool demand-off = Cool demand off
```
Comment (lolodomo) [inline bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/i18n/somfytahoma.properties:258]: Fix all options with space in entry

File: bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/i18n/somfytahoma.properties [hunk 8]
```diff
@@ -245,8 +323,54 @@ channel-type.somfytahoma.target_temperature.label = Target Temperature
...
+channel-type.somfytahoma.yutaki_target_mode.state.option.comfort = Comfort
+channel-type.somfytahoma.yutaki_target_mode.state.option.eco = Eco
+channel-type.somfytahoma.zone_mode.label = Zone Mode
+channel-type.somfytahoma.zone_mode.description = Zone mode control
+channel-type.somfytahoma.zone_mode.state.option.auto = Auto
+channel-type.somfytahoma.zone_mode.state.option.manu = Manual
+
+# channel types
+
+channel-type.somfytahoma.derogation_heating_mode.command.option.frost\ protection = Frost protection
```
Comment (lolodomo) [inline bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/i18n/somfytahoma.properties:376]: To be moved again at the right place.

**Review by lolodomo — CHANGES_REQUESTED** [review]
i18n tool does not support properly option value containing space.

File: bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/i18n/somfytahoma.properties (hunk 5 above)
Comment (octa22) [reply bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/i18n/somfytahoma.properties]: fixed

File: bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/i18n/somfytahoma.properties (hunk 6 above)
Comment (octa22) [reply bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/i18n/somfytahoma.properties]: fixed

File: bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/i18n/somfytahoma.properties (hunk 7 above)
Comment (octa22) [reply bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/i18n/somfytahoma.properties]: all fixed

File: bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/i18n/somfytahoma.properties (hunk 8 above)
Comment (octa22) [reply bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/i18n/somfytahoma.properties]: fixed

File: bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/thing/channels.xml (hunk 1 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/thing/channels.xml:757]: I mean you could replace
`<channel id="outdoor_temp" typeId="outdoor_temp"></channel>`
By
`<channel id="outdoor_temp" typeId="system.outdoor-temperature"></channel>`
And then remove outdoor_temp

Comment (octa22) [reply bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/thing/channels.xml]: thanks, done

**Review by lolodomo — APPROVED** [review]
LGTM

**Pull request merged**